src/main/java/org/jabref/gui/SendAsEMailAction.java

```diff
@@ -107,19 +111,30 @@ private String sendEmail() throws Exception {
             }
         }
 
-        String mailTo = "?Body=".concat(rawEntries.toString());
-        mailTo = mailTo.concat("&Subject=");
-        mailTo = mailTo.concat(preferencesService.getExternalApplicationsPreferences().getEmailSubject());
-        for (String path : attachments) {
-            mailTo = mailTo.concat("&Attachment=\"").concat(path);
-            mailTo = mailTo.concat("\"");
-        }
-
-        URI uriMailTo = new URI("mailto", mailTo, null);
+        URI uriMailTo = getUriMailTo(rawEntries, attachments);
 
         Desktop desktop = Desktop.getDesktop();
         desktop.mail(uriMailTo);
 
         return String.format("%s: %d", Localization.lang("Entries added to an email"), entries.size());
     }
+
+    private URI getUriMailTo(StringWriter rawEntries, List<String> attachments) throws URISyntaxException {
+        StringBuilder mailTo = new StringBuilder();
+        if (action == StandardActions.SEND_TO_KINDLE) {
+            mailTo.append(preferencesService.getExternalApplicationsPreferences().getKindleEmail());
+            mailTo.append("?Subject=").append(Localization.lang("Send to Kindle"));
+        } else if (action == StandardActions.SEND_AS_EMAIL) {
+            mailTo.append("?Body=").append(rawEntries.toString());
+            mailTo.append("&Subject=");
+            mailTo.append(preferencesService.getExternalApplicationsPreferences().getEmailSubject());
+        }
```

Comment:
Just a comment:

This is not object-oriented. Normally, one would crreate an abstract class and two sub classes, where the subclasses differ in that behavior.

However, these are 5 lines of Code only, so OK for me.